go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -271,25 +271,51 @@ func (c *Controller) syncServices(key string) error {
 
 	vips := collectServiceVIPs(service)
 	if svcNeedsIdling(service.Annotations) && !util.HasValidEndpoint(service, endpointSlices) {
+		// addServiceToIdlingBalancer adds the vips to service tracker
 		err = c.addServiceToIdlingBalancer(vips, service)
 		if err != nil {
 			c.eventRecorder.Eventf(service, v1.EventTypeWarning, "FailedToAddToIdlingBalancer",
 				"Error trying to add to Idling LoadBalancer for Service %s/%s: %v", name, namespace, err)
 			return err
 		}
-		vipsTracked = vipsTracked.Delete(vips.List()...)
-		err = deleteVIPsFromNonIdlingOVNBalancers(vips, name, namespace)
+		toRemoveFromNonIdling := sets.NewString()
+		for vipProtocol := range vips {
+			vip, _ := splitVirtualIPKey(vipProtocol)
+			if c.serviceTracker.getLoadBalancer(name, namespace, vip) != loadbalancer.IdlingLoadBalancer {
+				toRemoveFromNonIdling.Insert(vip)
```

Comment:
@fedepaol helped me figure out this is causing a segfault cause I'm passing the vip here and not vipProtocol. That is why the unidling tests are failing. will fix.